### PR TITLE
config_param options to specify deprecated/obsoleted parameters

### PR DESCRIPTION
**Pull request by tagomoris**
This change adds 2 options for `config_param`.
And I marked `host`, `port` of `out_forward` without `<server>` sections as obsoleted.

### deprecated
Unrecommended parameter, and will be removed in future.
If users specify this parameter, Fluentd warns it in logs.

### obsoleted
Configuration parameter which is already removed.
If users specify this parameter, Fluentd stops with an error (a kind of configuration errors).

### CODE REVIEW

**Comment by tagomoris** [comment]
@repeatedly could you review this change?

File: lib/fluent/config/section.rb [hunk 1]
```diff
@@ -138,6 +139,20 @@ def self.generate(proxy, conf, logger, plugin_class, stack = [])
...
             section_params[varname] = self.instance_exec(val, opts, name, &block)
+
+            # Source of definitions of deprecated/obsoleted:
+            # https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Deprecated_and_obsolete_features
+            #
+            # Deprecated: These deprecated features can still be used, but should be used with caution
+            #             because they are expected to be removed entirely sometime in the future.
+            # Obsoleted: These obsolete features have been entirely removed from JavaScript and can no longer be used.
+            if opts[:deprecated]
+              logger.warn "'#{name}' paramenter isn't recommended to use now."
```
Comment (repeatedly) [inline lib/fluent/config/section.rb]: typo: parameter

**Comment by repeatedly** [comment]
One idea. How about adding "message" like parameter to show better logs?.

```rb
 config_param :host, :string, default: nil, obsoleted: true, message: "Use <server> section instead"
```

result:

```
 'host' parameter is already removed. Don't use it. Use <server> section instead"
```

**Comment by tagomoris** [comment]
@repeatedly When is that parameter used? only for `deprecated` and `obsoleted`?
If so, it may be better to change value type of these options from true/false to string of message to show. Isn't it?

**Comment by repeatedly** [comment]
> it may be better to change value type of these options from true/false to string of message to show. Isn't it?

LGTM 👍

**Comment by repeatedly** [comment]
Good. After test passed, please merge.

**Comment by tagomoris** [comment]
Merged.

**Pull request merged**